doc: modify README and upgrade docs (#3933)
ceph/ceph-csi: docs/ceph-csi-upgrade.md

# Ceph-csi Upgrade

- [Ceph-csi Upgrade](#ceph-csi-upgrade)
  - [Pre-upgrade considerations](#pre-upgrade-considerations)
    - [Snapshot-controller and snapshot crd](#snapshot-controller-and-snapshot-crd)
      - [Snapshot API version support matrix](#snapshot-api-version-support-matrix)
  - [Upgrading from v3.2 to v3.3](#upgrading-from-v32-to-v33)
  - [Upgrading from v3.3 to v3.4](#upgrading-from-v33-to-v34)
  - [Upgrading from v3.4 to v3.5](#upgrading-from-v34-to-v35)
  - [Upgrading from v3.5 to v3.6](#upgrading-from-v35-to-v36)
  - [Upgrading from v3.6 to v3.7](#upgrading-from-v36-to-v37)

>   - [Upgrading from v3.7 to v3.8](#upgrading-from-v37-to-v38)

Reviewer: A small suggestion:
Instead of having seperate bullet points or headings for each of these older releases, I thik it is better to have it under one heading like **Upgrading from deprecated versions**. It will make the doc look cleaner, instead of having seperate headings for each release verion which has just a single line to refer to another section. With this, all the older release upgrade steps can be listed under one heading with a list of references to the relevant sections. Otherwise in the future, this doc would be filled with such a long list of upgrade headings and their reference links. I feel the current number of such headings itself is already long.

Author: done

- Drain your application pods from the node
- Delete the CSI driver pods on the node
   - The pods to delete will be named with a csi-cephfsplugin prefix and have a
     random suffix on each node. However, no need to delete the provisioner
     pods: csi-cephfsplugin-provisioner-* .
   - The pod deletion causes the pods to be restarted and updated automatically
     on the node.

##### 2.4 Modifying MountOptions in Storageclass and PersistentVolumes

CephCSI, starting from release v3.9.0, will pass the options specified in the

> StorageClass's `MountOptions` during both `NodeStage` (kernel cephfs or
> ceph-fuse mount operation) and `NodePublish` (bind mount) operations.

Reviewer 2: ```suggestion
StorageClass's `MountOptions` during both `NodeStageVolume` (kernel cephfs or
ceph-fuse mount operation) and `NodePublishVolume` (bind mount) operations.
```

Therefore, only common options that is acceptable during both the above

> described operation needs to be set in StorageClass's `MountOptions`.

Reviewer: s/operation/operations

If invalid mount options are set in StorageClass's `MountOptions`

> such as `"debug"`, the mounting of cephfs PVCs will fail.

Reviewer 2: ```suggestion
such as `"debug"`, the mounting of cephFS PVCs will fail.
```

Follow the below steps to update the StorageClass's `MountOptions`:

- Take a backup of the StorageClass using
  `kubectl get sc <storageclass-name> -o yaml > sc.yaml`.
- Edit `sc.yaml` to remove the invalid mount options from `MountOptions` field.
- Delete the StorageClass using `kubectl delete sc <storageclass-name>`.
- Recreate the StorageClass using `kubectl create -f sc.yaml`.

Follow the below steps to update the PersistentVolume's `MountOptions`:

> - Identify cephfs PersistentVolumes using

Reviewer 2: ```suggestion
Identify cephFS PersistentVolumes using
```
